File: 1/1608/SpecialArrayWithXElementsGreaterThanEqualToX.py
#   https://leetcode.com/problems/special-array-with-x-elements-greater-than-or-equal-x/description/?envType=daily-question&envId=2024-05-27

class Solution:
    def specialArray(self, nums: List[int]) -> int:

        def checkNumbersInNumsGreaterThanEqualToCurrentNumber(currentNumber, nums):
            lengthOfNums=len(nums)
            countOfNumbersGreaterThanEqualToCurrentNumber=0
            for i in range(0, lengthOfNums, 1):
                if nums[i]>=currentNumber:
                    countOfNumbersGreaterThanEqualToCurrentNumber+=1
            if countOfNumbersGreaterThanEqualToCurrentNumber==currentNumber:
                return True
            return False

        nums.sort()
        lengthOfNums=len(nums)
        lastNumberToCheck=nums[lengthOfNums-1]
        for currentNumber in range(0, lastNumberToCheck+1, 1):
            if checkNumbersInNumsGreaterThanEqualToCurrentNumber(currentNumber, nums):
                return currentNumber
        return -1

File: 1/1000s/16s/1608/SpecialArrayWithXElementsGreaterThanEqualToX.java
/*
https://leetcode.com/problems/special-array-with-x-elements-greater-than-or-equal-x/submissions/1269613113/?envType=daily-question&envId=2024-05-27
*/

class Solution {

    public boolean checkNumbersInNumsGreaterThanEqualToCurrentNumber(int currentNumber, int[] nums){
        int lengthOfNums=nums.length, countOfNumbersGreaterThanEqualToCurrentNumber=0;
        for(int i=0; i<lengthOfNums; ++i){
            if(nums[i]>=currentNumber){
                countOfNumbersGreaterThanEqualToCurrentNumber++;
            }
        }
        if(countOfNumbersGreaterThanEqualToCurrentNumber==currentNumber) return true;
        return false;
    }

    public int specialArray(int[] nums) {
        Arrays.sort(nums);
        int lengthOfNums=nums.length;
        int lastNumberToCheck=nums[lengthOfNums-1];
        for(int currentNumber=0; currentNumber<=lastNumberToCheck; ++currentNumber){
            if(checkNumbersInNumsGreaterThanEqualToCurrentNumber(currentNumber, nums)){
                return currentNumber;
            }
        }
        return -1;
    }
}